roles/openshift_facts/library/openshift_facts.py

```diff
@@ -239,49 +272,56 @@ def walk_metadata(self, metadata_url, headers=None, expect_json=False):
         for line in self.query_metadata(metadata_url, headers, expect_json):
             if line.endswith('/') and not line == 'public-keys/':
                 key = line[:-1]
-                metadata[key]=self.walk_metadata(metadata_url + line, headers,
-                                                 expect_json)
+                metadata[key] = self.walk_metadata(metadata_url + line,
+                                                   headers, expect_json)
             else:
                 results = self.query_metadata(metadata_url + line, headers,
                                               expect_json)
                 if len(results) == 1:
+                    # disable pylint maybe-no-member because overloaded use of
+                    # the module name causes pylint to not detect that results
+                    # is an array or hash
+                    # pylint: disable=maybe-no-member
                     metadata[line] = results.pop()
                 else:
                     metadata[line] = results
         return metadata
 
     def get_provider_metadata(self, metadata_url, supports_recursive=False,
-                          headers=None, expect_json=False):
+                              headers=None, expect_json=False):
         try:
             if supports_recursive:
-                metadata = self.query_metadata(metadata_url, headers, expect_json)
+                metadata = self.query_metadata(metadata_url, headers,
+                                               expect_json)
             else:
-                metadata = self.walk_metadata(metadata_url, headers, expect_json)
-        except OpenShiftFactsMetadataUnavailableError as e:
+                metadata = self.walk_metadata(metadata_url, headers,
+                                              expect_json)
+        except OpenShiftFactsMetadataUnavailableError:
             metadata = None
         return metadata
 
-    def get_ansible_facts(self):
-        if not hasattr(self, 'ansible_facts'):
-            self.ansible_facts = ansible_facts(module)
-        return self.ansible_facts
-
+    # TODO: refactor to reduce the size of this method, potentially create
+    # sub-methods (or classes for the different providers)
+    # temporarily disable pylint too-many-statements
+    # pylint: disable=too-many-statements
```

Comment:
@detiber I'm concerned with the number of pylint disables you're using in this file. I'm especially concerned because it seems that with each PR, the number of pylint disables keeps growing.

This specific one seems like it'd be relatively easy to fix by simply creating a few more methods and calling them.

I think we need to use the pylint disables only when absolutely necessary. That doesn't seem to be the case here.